utils/utils.ts

```diff
@@ -4,30 +4,81 @@
 
 interface IOptions {
```

Comment:
this seems like Options type, lets remove this and user the other one that is more easy to compose with other props